Doppler.Sap/Factory/SapTaskHandler.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Doppler.Sap.Models;
+using Newtonsoft.Json;
+using Microsoft.Extensions.Options;
+
+namespace Doppler.Sap.Factory
+{
+    public abstract class SapTaskHandler
+    {
+        protected readonly SapConfig SapConfig;
+        protected SapLoginCookies SapCookies;
+        private DateTime _sessionStartedAt;
+        protected readonly HttpClient Client;
+
+        protected SapTaskHandler(IOptions<SapConfig> sapConfig)
+        {
+            SapConfig = sapConfig.Value;
+
+            var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator,
+                UseCookies = false
+            };
+
+            Client = new HttpClient(handler);
+        }
+
+        public abstract Task<SapTaskResult> Handle(SapTask dequeuedTask);
+
+        protected async Task StartSession()
+        {
+            if (SapCookies == null || DateTime.Now > _sessionStartedAt.AddMinutes(30))
+            {
+                var sapResponse = await Client.SendAsync(new HttpRequestMessage
+                {
+                    RequestUri = new Uri($"{SapConfig.BaseServerURL}Login/"),
+                    Content = new StringContent(JsonConvert.SerializeObject(
```

Comment:
I am not sure, but probably exists something as JsonContent:

```suggestion
                    Content = new JsonContent(
```